### PR TITLE
[GPU] Move bitcast-convert expansion past layout assignment.

### PR DESCRIPTION
**Pull request by sergachev**
This lets post-layout assignment algebraic simplifier replace no-op bitcast-converts with bitcasts first.

### CODE REVIEW

**Review by akuegel — APPROVED** [review]

**Comment by akuegel** [comment]
@sergachev We see a presubmit failure for a jax test: https://github.com/jax-ml/jax/blob/main/tests/lax_numpy_test.py
Two tests are failing with your change:

LaxBackedNumpyTests.testView41 (shape=(2, 32), a_dtype=<class 'numpy.complex128'>, dtype=<class 'jax.numpy.int4'>)
LaxBackedNumpyTests.testView75 (shape=(2, 32), a_dtype=<class 'numpy.complex128'>, dtype=<class 'numpy.float32'>)

Can you please check whether you can reproduce this?

**Comment by sergachev** [comment]
Thanks! I couldn't run exact same tests but this gave me an idea what's wrong. I'll fix the layout normalization first.

**Review by akuegel — APPROVED** [review]
Confirmed that the Jax test is passing now with the adjustment of LayoutAssignment.
